x-pack/plugins/actions/server/builtin_action_types/lib/get_custom_agents.ts

```diff
@@ -33,6 +34,28 @@ export function getCustomAgents(
     return defaultAgents;
   }
 
+  let targetUrl: URL;
+  try {
+    targetUrl = new URL(url);
+  } catch (err) {
+    logger.warn(`error determining proxy state for invalid url "${url}", using default agents`);
```

Comment:
can we make sure this doesn't have auth info?

Reply:
we're not _using_ any auth info here that might be in the URL, and we _could_ fail here if we do have auth info in the URL, but ... it doesn't seem like the right place for it.  

Basically, the URL will end up coming from a connector config or secret (for Slack, since the URL has an API token at the end), so it seems like we should be rejecting URLs used as parameters for connectors that contain auth, in the connector validation, when connectors are created (it's also run when they are executed).  That way we'd know it would be auth-free at this point, which is way deep in the guts of the connector execution logic.

I think we'd want a new issue opened for that, will hit most of our connectors ...

Reply:
I opened https://github.com/elastic/kibana/issues/96236 for this, LMK if I got the intention wrong on it.